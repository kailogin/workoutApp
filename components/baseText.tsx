import React from "react";
import { StyleSheet, Text, TextStyle } from "react-native";

import { Colors } from "../utils/colors";

interface BaseTextProps {
  style?: TextStyle;
}

export const BaseText: React.FC<BaseTextProps> = ({ children, style }) => {
  return <Text style={[styles.text, style]}>{children}</Text>;
};

const styles = StyleSheet.create({
  text: {
    color: Colors.WHITE,
    fontFamily: "Montserrat",
  },
});
