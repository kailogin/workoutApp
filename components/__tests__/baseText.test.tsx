import "react-native";
import React from "react";
import renderer from "react-test-renderer";
import { FlatList } from "react-native";

import { BaseText } from "../baseText";

const TEXT = "Test Text";

test(`BaseText snapshot`, () => {
  const snap = renderer.create(<BaseText>{TEXT}</BaseText>).toJSON();

  expect(snap).toMatchSnapshot();
});

it("renders the Flatlist component with BaseText as renderItem", () => {
  const tree = renderer
    .create(
      <FlatList
        data={["Item1", "Item2", "Item3"]}
        keyExtractor={(item) => item}
        renderItem={({ item }) => <BaseText>{item}</BaseText>}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
