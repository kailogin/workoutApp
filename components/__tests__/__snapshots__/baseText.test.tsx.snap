// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BaseText snapshot 1`] = `
<Text
  style={
    Array [
      Object {
        "color": "#fff",
        "fontFamily": "Montserrat",
      },
      undefined,
    ]
  }
>
  Test Text
</Text>
`;

exports[`renders the Flatlist component with BaseText as renderItem 1`] = `
<RCTScrollView
  data={
    Array [
      "Item1",
      "Item2",
      "Item3",
    ]
  }
  disableVirtualization={false}
  getItem={[Function]}
  getItemCount={[Function]}
  horizontal={false}
  initialNumToRender={10}
  keyExtractor={[Function]}
  maxToRenderPerBatch={10}
  onContentSizeChange={[Function]}
  onEndReachedThreshold={2}
  onLayout={[Function]}
  onMomentumScrollEnd={[Function]}
  onScroll={[Function]}
  onScrollBeginDrag={[Function]}
  onScrollEndDrag={[Function]}
  removeClippedSubviews={false}
  renderItem={[Function]}
  scrollEventThrottle={50}
  stickyHeaderIndices={Array []}
  updateCellsBatchingPeriod={50}
  viewabilityConfigCallbackPairs={Array []}
  windowSize={21}
>
  <View>
    <View
      onLayout={[Function]}
      style={null}
    >
      <Text
        style={
          Array [
            Object {
              "color": "#fff",
              "fontFamily": "Montserrat",
            },
            undefined,
          ]
        }
      >
        Item1
      </Text>
    </View>
    <View
      onLayout={[Function]}
      style={null}
    >
      <Text
        style={
          Array [
            Object {
              "color": "#fff",
              "fontFamily": "Montserrat",
            },
            undefined,
          ]
        }
      >
        Item2
      </Text>
    </View>
    <View
      onLayout={[Function]}
      style={null}
    >
      <Text
        style={
          Array [
            Object {
              "color": "#fff",
              "fontFamily": "Montserrat",
            },
            undefined,
          ]
        }
      >
        Item3
      </Text>
    </View>
  </View>
</RCTScrollView>
`;
